src/backend/distributed/executor/citus_custom_scan.c

```diff
@@ -175,6 +185,44 @@ CitusExecScan(CustomScanState *node)
  */
 static void
 CitusModifyBeginScan(CustomScanState *node, EState *estate, int eflags)
+{
+	CitusScanState *scanState = (CitusScanState *) node;
+	DistributedPlan *distributedPlan = scanState->distributedPlan;
+
+	/*
+	 * If we have not already copied the plan into this context, do it now.
+	 * Note that we could have copied the plan during CitusGenerateDeferredQueryStrings.
+	 */
+	if (GetMemoryChunkContext(distributedPlan) != CurrentMemoryContext)
+	{
+		distributedPlan = copyObject(distributedPlan);
+
+		scanState->distributedPlan = distributedPlan;
+	}
+
+	Job *workerJob = distributedPlan->workerJob;
+	List *taskList = workerJob->taskList;
+
+	/* prevent concurrent placement changes */
+	AcquireMetadataLocks(taskList);
+
+	/*
+	 * We are taking locks on partitions of partitioned tables. These locks are
+	 * necessary for locking tables that appear in the SELECT part of the query.
+	 */
+	LockPartitionsInRelationList(distributedPlan->relationIdList, AccessShareLock);
```

Comment:
Do we need this here? Seems like we take locks in AdaptiveExecutor.

Reply:
It looks reasonable to remove this. We acquire the same locks in both AdaptiveExecutor and TaskTracker.